fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java

```diff
@@ -696,14 +698,24 @@ default boolean isReady() {
      * @param check the check to be performed
      */
     public synchronized void addCommitCheck(@Nonnull CommitCheckAsync check) {
+        addCommitCheck(null, check);
+    }
+
+    public synchronized void addCommitCheck(@Nullable String name, @Nonnull CommitCheckAsync check) {
         while (!commitChecks.isEmpty()) {
             if (commitChecks.peek().isReady()) {
                 asyncToSync(FDBStoreTimer.Waits.WAIT_ERROR_CHECK, commitChecks.remove().checkAsync());
+                if (name != null) {
+                    namedCommitChecks.remove(name);
+                }
```

Comment:
I don't think this is the intended semantics. It look to me like, if any pending check is ready, check it and allow the named check to supersede what was there before, whatever state it was in. Which means if a pending check by the given name is not yet ready, the new check with that name is ignored.

Questions about the desired behavior:
* If a check by the given name is ready, is it checked or is that result ignored? Assuming the check passes, is the new check remembered as well?
* If a check by the given name is pending, does the new check replace it?

Reply:
In the current situation a new check with the same name would 
- if the check is ready it would be checked and returned then the new check is remembered
- If it is not ready the check is not replaced and the old one is remembered. 
I believe the solve here if we WANT to replace the old check would be to create a specific replaceCommitCheck function.

Reply:
@MMcM Added a session concept for lookup...  Hope that clarifies.